### PR TITLE
docs: import "Adding a SQL Function" codelab

### PR DESCRIPTION
**Pull request by benesch**
Import "Adding a SQL Function", our first of hopefully many codelabs.

Codelabs are tutorials on how to develop for CockroachDB, in the style of Google Codelabs (https://codelabs.developers.google.com).

---

This is a conversion of the original Google Doc with essentially no content changes, save for a few last-minute copy edits. Someone should run through this once more tomorrow to catch any typos or bugs I may have introduced. I'm out of steam for the night!

### CODE REVIEW

**Comment by benesch** [comment]
For future reference, this is the rather half-baked script I used to do most of the conversion: https://github.com/mangini/gdocs2md

File: docs/codelabs/00-sql-function.md [hunk 1]
```diff
@@ -0,0 +1,387 @@
+# Codelab: Adding a SQL Function
+
+*Original author: Peter Mattis <<peter@cockroachlabs.com>>*
```
Comment (petermattis) [inline docs/codelabs/00-sql-function.md]: Either omit this author line, or add yourself as well.

File: docs/codelabs/00-sql-function.md [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+execution engine has its own typing system. Each type in the system adheres to
+the ``Datum`` interface which defines the methods that a type needs to
+implement. ``DString`` is the implementation of ``Datum`` for the SQL ``string``
+type.
+
+```go
+type DString string
+```
+
+Note that `DString` implements the `Datum` interface, not `DString`. This is why
```
Comment (petermattis) [inline docs/codelabs/00-sql-function.md]: The first `DString` should be `*DString`.

File: docs/codelabs/00-sql-function.md [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+pointer (`fn`).
+
+The SQL execution engine finds the `Builtin` structure given the name of a
+function using the `Builtins` map:
+
+```go
+var Builtins = map[string][]Builtin{...}
+```
+
+Notice that this a map from `string` to a slice of `Builtin`. The slice is used
```
Comment (petermattis) [inline docs/codelabs/00-sql-function.md]: `s/this a map/this is a map/g`.

File: docs/codelabs/00-sql-function.md [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+
+```text
+query II
+SELECT LENGTH('Hello, 世界'), LENGTH(b'Hello, 世界')
+----
+9 13
+```
+
+The format is relatively straightforward. `query II` means "there’s a query on
+the next line that will return two `I`nteger columns; please check that that
```
Comment (petermattis) [inline docs/codelabs/00-sql-function.md]: `I`nteger renders oddly in markdown. Perhaps **I**nteger.

**Review by petermattis — APPROVED** [review]
LGTM

This looks great.

**Comment by benesch** [comment]
Addressed everything—merging as soon as this goes green!

**Pull request merged**